File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_w = 0
    max_h = 0
    
    for size in sizes:
        current_long = max(size[0], size[1])
        current_short = min(size[0], size[1])
        
        max_w = max(max_w, current_long)
        max_h = max(max_h, current_short)
        
    return max_w * max_h

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let maxWidth = 0;
    let maxHeight = 0;

    sizes.forEach(([w, h]) => {
        const [long, short] = w > h ? [w, h] : [h, w];
        
        if (long > maxWidth) maxWidth = long;
        if (short > maxHeight) maxHeight = short;
    });

    return maxWidth * maxHeight;
}